src/JSPropUpdater.ts

```diff
@@ -55,11 +66,49 @@ export class JSPropUpdater {
     animatedComponent: React.Component<unknown, unknown>
   ) {
     const viewTag = findNodeHandle(animatedComponent);
-    JSPropUpdater._tagToComponentMapping.delete(viewTag);
-    if (JSPropUpdater._tagToComponentMapping.size === 0) {
+    JSPropUpdaterPaper._tagToComponentMapping.delete(viewTag);
+    if (JSPropUpdaterPaper._tagToComponentMapping.size === 0) {
       this._reanimatedEventEmitter.removeAllListeners(
         'onReanimatedPropsChange'
       );
     }
   }
 }
+
+class JSPropUpdaterFabric implements JSPropUpdater {
+  private static _tagToComponentMapping = new Map();
+  private static isInitialized = false;
+
+  constructor() {
+    if (!JSPropUpdaterFabric.isInitialized) {
+      const updater = (viewTag: number, props: unknown) => {
+        const component =
+          JSPropUpdaterFabric._tagToComponentMapping.get(viewTag);
+        component?._updateFromNative(props);
+      };
+      runOnUIImmediately(() => {
+        'worklet';
+        global._jsPropsUpdater = (viewTag: number, props: unknown) => {
```

Comment:
Also, why can't we just install JS props updater in the main RN runtime and use it from C++ via `JSScheduler`?

Reply:
We could use this approach, but I believe it would introduce more problems. The issue lies in copying the `nonAnimatableProps` to the second runtime. Since it is a `jsi::Value`, we cannot simply copy it.

However, we rely on the `nonAnimatableProps` in the UI to check for any JS prop and determine if we need to schedule an operation to the React runtime. Additionally, we need the `nonAnimatableProps` on React runtime to apply changes to the view.

One proposed solution is to calculate the `nonAnimatableProps` twice. However, I consider this approach to be both costly and unnecessary.